homeassistant/util/__init__.py

```diff
@@ -32,6 +33,27 @@
 RE_SANITIZE_PATH = re.compile(r"(~|\.(\.)+)")
 
```

Comment:
These regular expressions are confusing, they match a lot of valid names.

Reply:
_regular expressions are confusing_ :smile: 
That might be, but that should be handled in a separate PR if it were to change.

Reply:
This matches `~` or `..` or `...` etc. Those are not valid when coming from a user.

Reply:
The command
```
wget https://github.com/github/linguist/compare/c3a414e..faf7c6f.diff
```
will create a file with two consecutive dots. Emacs backup files end with a tilde. Those are real-world examples of perfectly valid and safe file names generated by a user.

The interface (not the regular expression as such) in this PR is confusing because it is unclear why those names are disallowed.

On the other hand, absolute paths are valid which seems like a possible gotcha.

> That might be, but that should be handled in a separate PR if it were to change.

I disagree, it is an integral part of the new interface.

Reply:
Usually `../` is checked for ... with some caution:
 https://github.com/swisskyrepo/PayloadsAllTheThings/blob/master/Directory%20Traversal/README.md#bypass--replaced-by-